instrumentation/khttp-0.1/src/main/java/io/opentelemetry/auto/instrumentation/khttp/KHttpInstrumentation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.auto.instrumentation.khttp;
+
+import static io.opentelemetry.auto.tooling.ClassLoaderMatcher.hasClassesNamed;
+import static io.opentelemetry.auto.tooling.bytebuddy.matcher.AgentElementMatchers.safeHasSuperType;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isAbstract;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.auto.tooling.Instrumenter;
+import java.util.Map;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(Instrumenter.class)
+public class KHttpInstrumentation extends Instrumenter.Default {
+
+  public KHttpInstrumentation() {
+    super("khttp");
+  }
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderMatcher() {
+    return hasClassesNamed("khttp.KHttp");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return safeHasSuperType(named("khttp.KHttp"));
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+      packageName + ".KHttpHeadersInjectAdapter",
+      packageName + ".KHttpDecorator",
+      packageName + ".RequestWrapper",
+    };
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isMethod()
+            .and(not(isAbstract()))
+            .and(
+                named("request")
+                    .and(takesArgument(0, named("java.lang.String")))
+                    .and(takesArgument(1, named("java.lang.String")))
+                    .and(
+                        takesArgument(2, named("java.util.Map"))
+                            .and(returns(named("khttp.responses.Response"))))),
```

Comment:
this will format nicer if you don't nest the ands:
```suggestion
            .and(not(isAbstract()))
            .and(named("request"))
            .and(takesArgument(0, named("java.lang.String")))
            .and(takesArgument(1, named("java.lang.String")))
            .and(takesArgument(2, named("java.util.Map")))
            .and(returns(named("khttp.responses.Response"))),
```

Reply:
goJF does not agree with you on that :)

Reply:
did u notice that i moved around the parentheses? I removed the nesting, which makes goJF format it nicer

Reply:
Indeed...